android/src/main/java/io/grpc/android/AndroidChannelBuilder.java

```diff
@@ -297,6 +297,11 @@ public void enterIdle() {
       public void onAvailable(Network network) {
         delegate.enterIdle();
       }
+      @Override
```

Comment:
I'm not sure how `@TargetApi` works but probably we need to define and register a separate filter for API level 29+

```
/** Respond to changes in the default network. Only used on API levels 29+. */
@TargetApi(Build.VERSION_CODES.Q)
private class BlockStatusChangedCallback extends ConnectivityManager.NetworkCallback {
```

Reply:
Well, I think `@Override` is the only thing that would break if `onBlockedStatusChanged()` is not present in the base class. So this code may "just work" on older android versions.

Reply:
This **might** fix a old longstanding issue downstream: https://github.com/FirebaseExtended/flutterfire/issues/6749

Reply:
+1

This might just work, at least my test on a older version Android suggests nothing is broken.